addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -554,6 +554,23 @@ def test_mailing_unique_name(self):
         self.assertEqual(mailing_0.name, 'First subject (Mass Mailing created on 2022-01-02)',
             msg='The name should be back to first one')
 
+    @freeze_time('2022-01-02')
+    @patch.object(Cursor, 'now', lambda *args, **kwargs: datetime(2022, 1, 2))
+    def test_mailing_create_with_context(self):
+        """ Test that the default_name provided via context is ignored to prevent constraint violations."""
+        mailing_1, mailing_2 = self.env["mailing.mailing"].create([
+            {
+                "subject": "First subject",
+                "name": "Mailing",
+            }, {
+                "subject": "Second subject",
+                "name": "Mailing",
+            }])
+        self.assertEqual(mailing_1.name, 'Mailing')
+        self.assertEqual(mailing_2.name, 'Mailing [2]')
+        mailing_3 = self.env['mailing.mailing'].with_context({'default_name': 'Mailing'}).create({"subject": "Third subject"})
+        self.assertEqual(mailing_3.name, 'Third subject (Mass Mailing created on 2022-01-02)')
```

Comment:
I feel like the name generation should take into account the context, in such a way that this will result as "Mailing [3]" instead.